.devcontainer/docker-compose.yml

```diff
@@ -0,0 +1,38 @@
+version: '3'
+services:
+  tedge:
+    build:
+      context: .
+      dockerfile: Dockerfile
+      args: 
+        VARIANT: bullseye
+    environment:
+      - DOCKER_HOST=unix:///opt/run/docker.sock
+
+    # Option 1: Use privileged container (for dev purposes only!)
+    volumes:
+      # Mount docker socket to something other than /var/run as a
+      # a privileged container or with CAP_SYS_ADMIN overwrites this
+      - /var/run/docker.sock:/opt/run/docker.sock
+      - tedge-bashhistory:/commandhistory
+      - device-certs:/etc/tedge/device-certs
+      # Mounts the project folder to '/workspace'. While this file is in .devcontainer,
+      # mounts are relative to the first file in the list, which is a level up.
+      - ..:/workspace:cached
+    privileged: true
+
+    # Option 2: Only mount cgroup as readonly (but requires it to already exist)
```

Comment:
When is it recommended to use this option over Option. 1?

Reply:
Option 2 is slightly more secure as it has less capabilities, however the different is only marginal. But I wanted to leave the option there so users are aware of it. The second option does require a bit more configuration because you have to run it on a system with systemctl already installed, or create a fake `/sys/fs/cgroup` folder and assign the correct permissions to it.